modules/bird2.nix

```diff
@@ -38,128 +38,167 @@ in
         roa4 table dnroa4;
         roa6 table dnroa6;
 
-        ${lib.optionalString config.networking.dn42.stayrtr.enable ''
-          protocol rpki roa_dn42 {
-            roa4 { table dnroa4; };
-            roa6 { table dnroa6; };
-            remote 127.0.0.1;
-            port 8082;
-            refresh 600;
-            retry 300;
-            expire 7200;
-          }
-        ''}
-        ${lib.optionalString config.networking.dn42.roagen.enable ''
-          protocol static {
-            roa4 { table dnroa4; };
-            include "${config.networking.dn42.roagen.outputDir}/dn42-roa4.conf";
-          }
-          protocol static {
-            roa6 { table dnroa6; };
-            include "${config.networking.dn42.roagen.outputDir}/dn42-roa6.conf";
-          }
-        ''}
-
         include "${../resources/community_filter.conf}";
         include "${../resources/filters.conf}";
-      
-        router id ${cfg.routerId};
-        hostname "${config.networking.hostName}.${config.networking.domain}";
+      '';
 
-        protocol device {
-          scan time 10;
-        }
-   
-        protocol kernel kernel_4 {
-          scan time 20;
-
-          ipv4 {
-            import none;
-            export filter {
-              if source = RTS_STATIC then reject;
-              krt_prefsrc = ${cfg.addr.v4};
-              accept;
-            };
-          };
-        }
+      inherit (cfg) routerId;
 
-        protocol kernel kernel_6 {
-          scan time 20;
+      templates.bgp.dn42_peer = ''
+        local as ${builtins.toString cfg.as};
```

Comment:
Maybe we should store the as as a string

Reply:
Why? Its a 32bit unsigned integer

Reply:
or use apply toString but minor issue